index.js

```diff
@@ -0,0 +1,269 @@
+'use strict';
+let _ = require('lodash')
+  , redis = require('redis')
+  , modelQueue = false
+  , schemas = {};
+
+class ModelQueue {
```

Comment:
Yeah all the modelQueue / schemas cache stuff got a little messy because I was trying to avoid having to create a new Redis connection for every schema.  I also got worried about setting up multiple redis pubsub connections and losing performance to repetitive chatter.  Would love to discuss some options here because I wasn't super-happy about it, but seemed like the easiest thing to do at the time